core/blockchain/validate.go

```diff
@@ -355,24 +355,35 @@ func CheckTransactionSanity(tx *types.Transaction, params *params.Params) error
 		}
 		if len(tx.TxOut) >= 2 {
 			slen = len(tx.TxOut[1].PkScript)
-			if slen < MinCoinbaseScriptLen || slen > MaxCoinbaseScriptLen {
-				str := fmt.Sprintf("coinbase transaction script "+
-					"length of %d is out of range (min: %d, max: "+
-					"%d)", slen, MinCoinbaseScriptLen,
-					MaxCoinbaseScriptLen)
-				return ruleError(ErrBadCoinbaseScriptLen, str)
+			if params.HasTax() {
+				if slen < MinCoinbaseScriptLen || slen > MaxCoinbaseScriptLen {
+					str := fmt.Sprintf("coinbase transaction script "+
+						"length of %d is out of range (min: %d, max: "+
+						"%d)", slen, MinCoinbaseScriptLen,
+						MaxCoinbaseScriptLen)
+					return ruleError(ErrBadCoinbaseScriptLen, str)
+				}
+				orgPkScriptStr := hex.EncodeToString(params.OrganizationPkScript)
+				curPkScriptStr := hex.EncodeToString(tx.TxOut[1].PkScript)
+				if orgPkScriptStr != curPkScriptStr {
+					str := fmt.Sprintf("coinbase transaction for block pays to %s, but it is %s",
+						orgPkScriptStr, curPkScriptStr)
+					return ruleError(ErrBadCoinbaseValue, str)
+				}
+			} else {
+				if slen != 0 || tx.TxOut[1].Amount != 0 {
+					str := fmt.Sprintf("coinbase transaction error:no tax.")
+					return ruleError(ErrBadCoinbaseValue, str)
+				}
 			}
-			orgPkScriptStr := hex.EncodeToString(params.OrganizationPkScript)
-			curPkScriptStr := hex.EncodeToString(tx.TxOut[1].PkScript)
-			if orgPkScriptStr != curPkScriptStr {
-				str := fmt.Sprintf("coinbase transaction for block pays to %s, but it is %s",
-					orgPkScriptStr, curPkScriptStr)
-				return ruleError(ErrBadCoinbaseValue, str)
-			}
-
-		} else if len(tx.TxOut) >= 3 {
+		}
+		if len(tx.TxOut) >= 3 {
```

Comment:
The `if` statement results differently than the original `else if` statement, please do a double review of the change here.

Reply:
👌